### PR TITLE
Return error for no argument pagination

### LINKED ISSUE #747: Return error for no option pagination arguments passed into queries
If no pagination arguments are passed into a query it currently returns an empty list.  We should instead return an error.  See [here](https://github.com/FuelLabs/fuel-core/pull/508) for a similar pr.

### PR DESCRIPTION
**Pull request by ControlCplusControlV**
Fixes #747 https://github.com/FuelLabs/fuel-core/issues/382

Technically breaking but there shouldn't(?) be anything downstream which relies on getting nothing. This behavior is more of a bug than a feature/interface at this point

### CODE REVIEW

File: fuel-core/src/schema.rs
Comment (leviathanbeak) [inline fuel-core/src/schema.rs:45]: Can the name be more descriptive + potentially a comment above the function?
I know that it `checks` parameters but for what?, their validity, their combo-validity, etc?

**Comment by leviathanbeak** [comment]
looks good, will approve after @bvrooman 's suggestion is addressed (and potentially mine if @ControlCplusControlV  agrees with it)

File: fuel-core/src/schema.rs
Comment (xgreenx) [inline fuel-core/src/schema.rs:52]: What do you think about this variant? It seems more readable to me.

```rust
match (first, before, last, after) {
    (Some(_), Some(_), _, _)
    | (_, _, Some(_), Some(_))
    | (None, Some(_), _, Some(_))
    | (None, None, None, None) => true,
    (_, _, _, _) => false,
}
```

Also, do we need to allow `(None, Some(before), Some(last), None)` and `(Some(first), None, None, Some(after))`?

File: fuel-core/src/schema/balance.rs
Comment (xgreenx) [inline fuel-core/src/schema/balance.rs:123]: We have this check "(first.is_some() && before.is_some())" in 7 places, but you only updated it in 4
image

Maybe better to create a new function, `query` or `query_local`, and use it everywhere with this check inside=)

**Review by xgreenx — CHANGES_REQUESTED** [review]

File: fuel-core/src/schema.rs
Comment (ControlCplusControlV) [reply fuel-core/src/schema.rs:45]: Yes I can change that to be more descriptive

**Comment by ControlCplusControlV** [comment]
> looks good, will approve after @bvrooman 's suggestion is addressed (and potentially mine if @ControlCplusControlV agrees with it)

I do, with some of the changes surrounding moving out fuel-client I had been dragging my feet a bit, since `fuel-client` doesn't support empty pagination queries  and so this requires a new cynic version to test

File: fuel-core/src/schema/coin.rs
Comment (xgreenx) [inline fuel-core/src/schema/coin.rs:197]: I removed this filter because, on the SDK side, we use it to get all coins(spent and unspent).

**Review by Voxelot — APPROVED** [review]

**Review by xgreenx — APPROVED** [review]

**Comment by Voxelot** [comment]
Looks good but I'd like to get 0.15.1 released with the sequential slot access before merging this

**Pull request merged**
